moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -578,29 +578,29 @@ void IKFastKinematicsPlugin::getSolution(const IkSolutionList<IkReal>& solutions
 
 double IKFastKinematicsPlugin::harmonize(const std::vector<double>& ik_seed_state, std::vector<double>& solution) const
 {
-  double dist_abs = 0;
+  double dist_sqr = 0;
   std::vector<double> ss = ik_seed_state;
-  for (std::size_t i = 0; i < ik_seed_state.size(); ++i)
+  for (size_t i = 0; i < ik_seed_state.size(); ++i)
   {
-    while (ss[i] > M_PI)
+    while (ss[i] > 2 * M_PI)
     {
       ss[i] -= 2 * M_PI;
     }
-    while (ss[i] < -1 * M_PI)
+    while (ss[i] < 2 * M_PI)
     {
       ss[i] += 2 * M_PI;
     }
-    while (solution[i] > M_PI)
+    while (solution[i] > 2 * M_PI)
```

Comment:
Initially when I filed [this PR](https://github.com/ros-planning/moveit/pull/537), I had modified the harmonize function to be used in getPositionIK but I modified getPositionIK to remove its dependency on harmonize function. In current commit, It is the original [harmonize function](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp#L579) which is not in good condition as referred in [pull#53](https://github.com/ros-planning/moveit_ikfast/pull/53). 
So, harmonize isn't part of this PR anymore. It will be good if you can review the [current commit](https://github.com/ros-planning/moveit/pull/537/files).